tests/unit/test_schema_reader.py

```diff
@@ -515,71 +529,191 @@ def test_message_error_handling(
+                    ),
+                    schema_type=SchemaType.PROTOBUF,
+                    message_type=MessageType.schema,
+                    expected_exception=CorruptKafkaRecordException,
+                    expected_warn_message="Invalid Protobuf references",
+                ),
+            ],
+        ),
+        MultipleMessagesHandlingErrorTestCase(
+            test_name="Invalid AVRO schema because referencing a corrupted but syntactically valid JSON",
+            inner_test_cases=[
+                # Given an invalid AVRO schema (corrupted but syntactically valid JSON)
+                # When handling the corrupted schema
+                # Then the schema is recognised as invalid because corrupted
+                SingleMessageHandlingErrorTestCase(
+                    test_name="AVRO corrupted but valid JSON",
+                    key=b'{"keytype":"SCHEMA","subject":"testref","version":1,"magic":1}',
+                    value=(
+                        b'{"schemaType": "AVRO", "subject": "testref", "version": 1, "id": 1, "deleted": false'
+                        + b', "schema": '
+                        + schema_avro_corrupted_but_valid_json
+                        + b"}"
+                    ),
+                    schema_type=SchemaType.AVRO,
+                    message_type=MessageType.schema,
+                    # expected_exception=CorruptKafkaRecordException,
+                    expected_exception=None,
+                    # expected_warn_message='TODO',
+                    expected_warn_message=None,
+                ),
+                # And given a AVRO schema referencing that corrupted schema (valid otherwise)
+                # When handling the schema
+                # Then the schema is recognised as invalid because of the corrupted referenced schema
+                SingleMessageHandlingErrorTestCase(
+                    test_name="AVRO referencing corrupted but valid JSON",
+                    key=b'{"keytype":"SCHEMA","subject":"test","version":1,"magic":1}',
+                    value=(
+                        b'{"schemaType": "AVRO", "subject": "test", "version": 1, "id": 1, "deleted": false'
+                        + b', "schema": '
+                        + schema_avro_referencing_corrupted_but_valid_json
+                        + b', "references": [{"name": "testref.avsc", "subject": "testref", "version": 1}]'
+                        + b"}"
+                    ),
+                    schema_type=SchemaType.AVRO,
+                    message_type=MessageType.schema,
+                    expected_exception=CorruptKafkaRecordException,
+                    # expected_exception=None,
+                    # expected_warn_message='TODO',
+                    expected_warn_message=None,
+                ),
+            ],
+        ),
+        MultipleMessagesHandlingErrorTestCase(
+            test_name="Invalid AVRO schema because referencing a corrupted and syntactically invalid JSON",
```

Comment:
@nosahama  Can you also review exceptions raised and warnings logged in this test case too (e.g. `InvalidReferences` vs `CorruptKafkaRecordException`)

Reply:
Within scope of the schema reader, I think `InvalidReferences` should end up being raised as `CorruptKafkaRecordException` and then we shutdown if the strict mode is on.